### PR TITLE
MathML presentation printer: Added Methods for Indexed, IndexedBase and tuple type objects

### LINKED ISSUE #12089: mathml printer for IndexedBase required
Writing an `Indexed` object to MathML fails with a `TypeError` exception: `TypeError: 'Indexed' object is not iterable`:

```
In [340]: sympy.__version__
Out[340]: '1.0.1.dev'

In [341]: from sympy.abc import (a, b)

In [342]: sympy.printing.mathml(sympy.IndexedBase(a)[b])
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-342-b32e493b70d3> in <module>()
----> 1 sympy.printing.mathml(sympy.IndexedBase(a)[b])

/dev/shm/gerrit/venv/stable-3.5/lib/python3.5/site-packages/sympy/printing/mathml.py in mathml(expr, **settings)
    442 def mathml(expr, **settings):
    443     """Returns the MathML representation of expr"""
--> 444     return MathMLPrinter(settings).doprint(expr)
    445 
    446 

/dev/shm/gerrit/venv/stable-3.5/lib/python3.5/site-packages/sympy/printing/mathml.py in doprint(self, expr)
     36         Prints the expression as MathML.
     37         """
---> 38         mathML = Printer._print(self, expr)
     39         unistr = mathML.toxml()
     40         xmlbstr = unistr.encode('ascii', 'xmlcharrefreplace')

/dev/shm/gerrit/venv/stable-3.5/lib/python3.5/site-packages/sympy/printing/printer.py in _print(self, expr, *args, **kwargs)
    255                 printmethod = '_print_' + cls.__name__
    256                 if hasattr(self, printmethod):
--> 257                     return getattr(self, printmethod)(expr, *args, **kwargs)
    258             # Unknown object, fall back to the emptyPrinter.
    259             return self.emptyPrinter(expr)

/dev/shm/gerrit/venv/stable-3.5/lib/python3.5/site-packages/sympy/printing/mathml.py in _print_Basic(self, e)
    356     def _print_Basic(self, e):
    357         x = self.dom.createElement(self.mathml_tag(e))
--> 358         for arg in e:
    359             x.appendChild(self._print(arg))
    360         return x

TypeError: 'Indexed' object is not iterable
```

It also fails for more complex expressions where at least one element is Indexed.

### PR DESCRIPTION
**Pull request by pragyanmehrotra**
…ple type objects in the presentation printer

#### References to other Issues or PRs
Fixes #12089   

#### Brief description of what is fixed or changed
MathML support for presentation printing Indexed, IndexedBase and tuple has been added.

#### Release Notes

* printing
  * Added MathML presentation printer method for printing Indexed objects
  * Added MathML presentation printer method for printing IndexedBase objects
  * Added MathML presentation printer methods for printing tuple objects

### CODE REVIEW

File: sympy/printing/mathml.py [hunk 1]
```diff
@@ -1041,4 +1062,4 @@ def print_mathml(expr, printer='content', **settings):
     print(pretty_xml)
 
 #For backward compatibility
-MathMLPrinter = MathMLContentPrinter
+MathMLPrinter = MathMLContentPrinter
```
Comment (Abdullahjavednesar) [inline sympy/printing/mathml.py]: Add a new line.

**Comment by oscargus** [comment]
Thanks! I will have a closer look at it, but for now:

In the release notes, this should only go in printing (functions is a separate module, so even if you add functions here, only functions in functions goes there). Also, it makes sense to add to which printer this is added (MathML Presentation).

Some tests are also needed. See the file sympy/printing/tests/test_mathml.py to see how they can look (check towards the end).

File: sympy/printing/mathml.py [hunk 2]
```diff
@@ -1196,6 +1196,26 @@ def _print_EmptySet(self, e):
         return x
 
 
+    def _print_tuple(self,e):
```
Comment (oscargus) [inline sympy/printing/mathml.py]: Space before e. (Same below.)

**Comment by oscargus** [comment]
Looks good! With some tests and some spaces it should be good to go.

**Comment by pragyanmehrotra** [comment]
I'll add the spaces right away, should I add the tests as well?

File: sympy/printing/mathml.py (hunk 2 above)
Comment (pragyanmehrotra) [reply sympy/printing/mathml.py]: done!

**Comment by oscargus** [comment]
Yes, please!

**Comment by pragyanmehrotra** [comment]
Added a few tests for printing IndexedBase type objects, please have a look at it are they good enough? or should I add more? or any edits...

File: sympy/printing/mathml.py [hunk 3]
```diff
@@ -1271,6 +1271,25 @@ def _print_Lambda(self, e):
...
+    def _print_IndexedBase(self, e):
+        return self._print(e.label)
+
+    def _print_Indexed(self, e):
+        x = self.dom.createElement('msub')
+        x.appendChild(self._print(e.base))
+        if len(e.indices) == 1:
+            x.appendChild(self._print(e.indices[0]))
+            return x
+        x.appendChild(self._print(e.indices))
```
Comment (oscargus) [inline sympy/printing/mathml.py]: There should be a return here, right? Or is it that indices[0] is a tuple when there is more than one? (If the tests pass.) Apart from that this looks good and the new tests look good. Maybe one could also add explicit tests for `tuple` and possibly `Indexed` although I understand that they are covered by the current test.

**Comment by pragyanmehrotra** [comment]
oh sorry while deleting the last line while fixing the conflict i accidentally deleted the return x lemme fix it and add explicit tests for them as well

**Comment by pragyanmehrotra** [comment]
added test_print_Indexed for Indexed type objects, however since test_print_tuples already existed, so I preferred not to create a duplicate definition of the same

File: sympy/printing/mathml.py (hunk 3 above)
Comment (pragyanmehrotra) [reply sympy/printing/mathml.py]: added return x, Thanks for pointing that out.

**Comment by oscargus** [comment]
Looks good! Only remaning issue is what you just fixed so if the tests pass I will merge it. Thanks!

**Pull request merged**